src/rocksdb.rs

```diff
@@ -276,7 +278,16 @@ impl DB {
     	if cfs.len() != cf_opts.len() {
 			return Err(format!("Mismatching number of CF options"));
 		}
-        let cpath = match CString::new(path.as_bytes()) {
+        let encoded_path = match Encoding::ANSI.to_bytes(path) {
```

Comment:
idiomatic rust tip: you can use `map_err` and the `?` operator to condense this to
`let encoded_path = Encoding::ANSI.to_bytes(path).map_err(|_| "error msg".to_string())?`